File: 조준장/골드5/13549.py
"""
- dp
"""
import sys
from collections import deque


n, m = map(int, sys.stdin.readline().split())

queue = deque([[n, 0]])
visited = [False] * 100001
visited[n] = True

while queue:
    s, depth = queue.popleft()
    if s == m:
        print(depth)
        break
    else:

        if 0 <= s - 1 < 100001 and not visited[s - 1]:
            queue.append([s - 1, depth + 1])
            visited[s - 1] = True
        
        if 0 <= s * 2 < 100001 and not visited[s * 2]:
            queue.appendleft([s * 2, depth])
            visited[s * 2] = True

        if 0 <= s + 1 < 100001 and not visited[s + 1]:
            queue.append([s + 1, depth + 1])
            visited[s + 1] = True


  

File: 조준장/골드5/13549.kt
var n = 0
var k = 0
val answer = mutableListOf<Int>()

fun main() {
    val br = System.`in`.buffered().bufferedReader()
    val bw = System.out.buffered().bufferedWriter()
    val sb = StringBuilder()

    val input = br.readLine().split(" ").map { it.toInt() }

    n = input.first()
    k = input[1]

    bfs()
    sb.append(answer.min())
    bw.write(sb.toString())
    br.close()
    bw.close()


}

data class Node(
    val x: Int,
    val depth: Int,
)

fun bfs() {
    val deque = ArrayDeque<Node>()
    deque.add(Node(n, 0))
    val visited = mutableSetOf<Int>()


    while (deque.isNotEmpty()) {
        val (tx, tDepth) = deque.removeFirst()

        if (tx == k) {
            answer.add(tDepth)
            continue
        }


        if (tx - 1 !in visited && 0 <= tx && tx - 1 < 100001) {
            visited.add(tx - 1)
            deque.add(Node(tx - 1, tDepth + 1))

        }

        if (2 * tx !in visited && 0 <= tx && 2 * tx < 100001) {
            visited.add(2 * tx)
            deque.add(Node(tx * 2, tDepth))


        }

        if (tx + 1 !in visited && 0 <= tx && tx + 1 < 100001) {
            visited.add(tx + 1)
            deque.add(Node(tx + 1, tDepth + 1))
        }

    }
}
